x-pack/plugins/cases/public/components/all_cases/all_cases_list.tsx

```diff
@@ -150,23 +143,14 @@ export const AllCasesList = React.memo<AllCasesListProps>(
     const onFilterChangedCallback = useCallback(
       (newFilterOptions: Partial<FilterOptions>) => {
         if (newFilterOptions.status && newFilterOptions.status === CaseStatuses.closed) {
-          setQueryParams((prevQueryParams) => ({
-            ...prevQueryParams,
-            sortField: SortFieldCase.closedAt,
-          }));
-        } else if (newFilterOptions.status && newFilterOptions.status === CaseStatuses.open) {
-          setQueryParams((prevQueryParams) => ({
-            ...prevQueryParams,
-            sortField: SortFieldCase.createdAt,
-          }));
+          setQueryParams({ sortField: SortFieldCase.closedAt });
```

Comment:
I think we should keep the previous logic and do `setQueryParams((prevQueryParams) => { ...prevQueryParams, sortField: SortFieldCase.closedAt });`. The reason is that inside `persistAndUpdateQueryParams` we do

```
if (isModalView) {
        setQueryParams(params);
        return;
      }
```

and any change done previously will be removed. For example, If I open the modal and then change the per page option and then sort by the "Created" field the per page setting will be lost.

Reply:
Yea,h I removed it because we cannot do `(prevQueryParams) => {...}` anymore, but the new logic takes into account what used to be there before anyway. 😄 

What is missing is the `isModalView` scenario indeed. I will add that 👍

Reply:
Let's do it on every call of `setQueryParams` in the hook. What do you think?

Reply:
I think on the second call it is unnecessary because `newQueryParams` has type `QueryParams` where every field is required, but I can add it for consistency's sake.